evap/results/tests/test_views.py

```diff
@@ -264,6 +265,38 @@ def make_course_with_evaluations(unique_suffix):
         caches["sessions"].clear()
         caches["results"].clear()
 
+    def test_evaluation_weight_sums(self):
+        """
+        ensures that the sum of evaluation weights add up to 100%,
+        even if some evaluations are not shown to the user
+        """
+        student = baker.make(UserProfile, email="student@institution.example.com")
+        course = baker.make(Course)
+        evaluations = [
+            (
+                baker.make(
+                    Evaluation,
+                    course=course,
+                    name_en=f"evaluation{i}",
+                    name_de=f"evaluation{i}",
+                    state=Evaluation.State.PUBLISHED,
+                    weight=5 - i,  # weights 1 to 5
+                    is_single_result=True,
+                )
+            )
+            for i in range(0, 5)
+        ]
+        for evaluation in evaluations:
+            baker.make(RatingAnswerCounter, contribution=evaluation.general_contribution, answer=2, count=2)
+        warm_up_template_cache(evaluations)
+        page = self.app.get(self.url, user=student).body.decode()
+        percentages = [6, 13, 20, 26, 33]
+        # percentage are floored, see evaluation/templatetags/evaluation_filters.py:100
+        prev_index = 0
+        for i in range(0, 5):
+            next_index = page.index(str(percentages[i]) + "%", prev_index)
```

Comment:
You pass prev_index here as a search start, so next_index would always have to be bigger than that. I'd just remove that argument. While at it, you could also use f-strings here.
```suggestion
            next_index = page.index(f"{percentages[i]}%")
```